integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/End2EndTestITStep.java

```diff
@@ -0,0 +1,272 @@
+    private static ProcessorRequest processorRequest;
+
+    @Given("Manager url is not accessible")
+    public void authenticationIsEnabled() {
+        given()
+                .filter(new ResponseLoggingFilter())
+                .contentType(ContentType.JSON)
+                .when()
+                .get(managerUrl + APIConstants.USER_API_BASE_PATH)
+                .then()
+                .statusCode(401);
+    }
+
+    @Given("Bridge list is empty")
+    public void getEmptyBridges() {
+
+        BridgeListResponse response = getBridgeList();
+        assertThat(response.getKind()).isEqualTo("BridgeList");
+        assertThat(response.getSize()).isZero();
+        assertThat(response.getPage()).isZero();
+        assertThat(response.getTotal()).isZero();
+    }
+
+    @When("Create a Bridge")
+    public void createBridge() {
+        BridgeResponse response = addBridge(BRIDGE_NAME);
+        assertThat(response.getName()).isEqualTo(BRIDGE_NAME);
+        assertThat(response.getStatus()).isEqualTo(BridgeStatus.REQUESTED);
+        assertThat(response.getEndpoint()).isNull();
+        assertThat(response.getPublishedAt()).isNull();
+        assertThat(response.getHref()).isNotNull();
+        assertThat(response.getSubmittedAt()).isNotNull();
+
+        bridgeId = response.getId();
+    }
+
+    @Then("^Bridge is exists in status \"([^\"]*)\" within (\\d+) (?:minute|minutes)$")
+    public void bridgeDeployedWithinMinutes(String status, int timeoutMinutes) {
+        Awaitility.await()
+                .atMost(Duration.ofMinutes(timeoutMinutes))
+                .pollInterval(Duration.ofSeconds(5))
+                .untilAsserted(
+                        () -> getBridgeDetails(bridgeId)
+                                .then()
+                                .body("status", Matchers.equalTo(status))
+                                .body("endpoint", Matchers.containsString(bridgeId)));
+
+        // store bridge endpoint details
+        endPoint = getBridgeDetails(bridgeId).then().extract().response().as(BridgeResponse.class).getEndpoint();
+    }
+
+    @When("Add processor to the bridge")
+    public void addProcessor() {
+
+        processorRequest = getProcessorRequest(FILTERS);
+
+        ProcessorResponse response = createProcessor(bridgeId, processorRequest);
+
+        processorId = response.getId();
+
+        assertThat(response.getName()).isEqualTo(PROCESSOR_NAME);
+        assertThat(response.getKind()).isEqualTo("Processor");
+        assertThat(response.getHref()).isNotNull();
+        assertThat(response.getStatus()).isEqualTo(BridgeStatus.REQUESTED);
+        assertThat(response.getFilters().size()).isEqualTo(1);
+
+        BaseAction action = response.getAction();
+        assertThat(action.getType()).isEqualTo(KafkaTopicAction.TYPE);
+        assertThat(action.getParameters()).containsEntry(KafkaTopicAction.TOPIC_PARAM, TOPIC_NAME);
+    }
+
+    @Then("Add wrong filter processor")
```

Comment:
@When("^add Processor to the Bridge with access token returns HTTP response code 400:$")
(with malformed request body in feature file)